name.abuchen.portfolio/src/name/abuchen/portfolio/datatransfer/pdf/FidelityPDFExtractor.java

```diff
@@ -0,0 +1,216 @@
+import name.abuchen.portfolio.model.Client;
+import name.abuchen.portfolio.model.PortfolioTransaction;
+import name.abuchen.portfolio.model.Transaction.Unit;
+import name.abuchen.portfolio.money.Money;
+import name.abuchen.portfolio.money.Values;
+
+@SuppressWarnings("nls")
+public class FidelityPDFExtractor extends AbstractPDFExtractor
+{
+    public FidelityPDFExtractor(Client client)
+    {
+        super(client);
+
+        addBankIdentifier("FIDELITY"); //$NON-NLS-1$
+
+        addSummaryStatementBuySellTransaction();
+        addBuySellTransaction();
+    }
+
+    private void addSummaryStatementBuySellTransaction()
+    {
+        // line must start with this!
+        DocumentType type = new DocumentType("Transaction Confirmation");
+        this.addDocumentTyp(type);
+
+        Block block = new Block("REFERENCE NO\\. TYPE REG\\.REP\\. TRADE DATE SETTLEMENT DATE CUSIP NO\\. ORDER NO\\.",
+                        "^(?!REFERENCE)(?!DESCRIPTION)[A-Z]{2,5}.*$");
+        type.addBlock(block);
+        block.set(new Transaction<BuySellEntry>()
+
+                        .subject(() -> {
+                            BuySellEntry entry = new BuySellEntry();
+                            entry.setType(PortfolioTransaction.Type.BUY);
+                            return entry;
+                        })
+                        /* @formatter:off
+                         * REFERENCE NO. TYPE REG.REP. TRADE DATE SETTLEMENT DATE CUSIP NO. ORDER NO.
+                         * 20123-1XXXXX 1* WK# 01-04-21 01-06-21 46428Q109 20123-XXXXX
+                         * DESCRIPTION and DISCLOSURES
+                         * You Bought ISHARES SILVER TR ISHARES Principal Amount       1,011.60
+                         *        40 WE HAVE ACTED AS AGENT. Settlement Amount       1,011.60
+                         *               at    25.2900 FBS receives compensation from the fund's advisor or its affiliates in connection with an
+                         * Symbol: exclusive, long-term marketing program that includes the promotion of this security and
+                         * SLV other iShares funds, and the inclusion of iShares funds in certain platforms and investment
+                         * @formatter:on
+                         */
+
+                        .section("date") //
+                        .match("^.* \\w{2}# (?<date>[\\d]{2}\\-[\\d]{2}\\-[\\d]{2}) .*$")
+                        .assign((t, v) -> t.setDate(asDate(v.get("date"))))
+
+                        .section("type", "gross", "tickerSymbol", "name") //
+                        .match("^You (?<type>Bought|Sold) (?<name>.*) Principal Amount\\s+(?<gross>[\\.,\\d]+)$") //
+                        .match("^(?!REFERENCE)(?!DESCRIPTION)(?<tickerSymbol>[A-Z]{2,5}).*$") //
+                        .assign((t, v) -> {
+                            t.setCurrencyCode(asCurrencyCode("USD"));
+                            v.put("currency", "USD");
```

Comment:
You can use the "$" to identify the currency. You can find the appropriate regex -> [here](https://github.com/portfolio-performance/portfolio/blob/master/CONTRIBUTING.md#regular-expressions) <-.

Reply:
there is no $ sign in the text :(